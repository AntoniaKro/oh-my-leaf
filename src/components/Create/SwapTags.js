import Label from '../../components/common/FormElements/Label';
import React from 'react';
import styled from 'styled-components';
import { SwapTag } from '../common/FormElements/Tags';

const StyledTagInputWrapper = styled.section`
  display: flex;
  flex-direction: column;
`;

const StyledTags = styled.div`
  margin: 5px 0;
  display: flex;
  flex-wrap: wrap;
`;

const StyledDelete = styled.b`
  margin-left: 3px;
`;

const StyledTagInput = styled.div`
  border: 2px solid #abc38e;
  border-radius: 11px;
  display: inline-block;
  font-family: 'PT Mono', monospace;
  font-size: 15px;
  padding: 5px;
`;

const StyledTagInputField = styled.input`
  border: 0;
  outline: 0;
  padding: 5px;
  font-size: 15px;
  font-family: 'PT Mono', monospace;
`;

function SwapTags({ tags, onDelete, onInput }) {
  return (
    <StyledTagInputWrapper>
      <Label htmlFor="swaps">Swap against (seperate by comma)</Label>
      <StyledTagInput>
        <StyledTags>
          {tags.map(tag => (
            <SwapTag key={tag}>
              {tag} <StyledDelete onClick={() => onDelete(tag)}>x</StyledDelete>
            </SwapTag>
          ))}
        </StyledTags>
        <StyledTagInputField
          id="swaps"
          name="swaps"
          placeholder="type new tag here..."
          onInput={event => event.target.value.includes(',') && onInput(event)}
        />
      </StyledTagInput>
    </StyledTagInputWrapper>
  );
}

export default SwapTags;
