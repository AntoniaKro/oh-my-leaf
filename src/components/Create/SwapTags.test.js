import React from 'react';
import SwapTags from './SwapTags';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const tags = ['test'];

describe('Swap tags component', () => {
  it('renders a section with an input field to add tags to a tag list', () => {
    const input = renderer.create(<SwapTags tags={tags} />);
    const tree = input.toJSON();
    expect(tags).toMatchSnapshot();
  });
});
/* tags on Delete onInput*/
