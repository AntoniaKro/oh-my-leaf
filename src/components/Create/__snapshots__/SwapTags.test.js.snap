// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swap tags component renders a section with an input field to add tags to a tag list 1`] = `
<section
  className="SwapTags__StyledTagInputWrapper-sc-15dr4e3-0 eSPwwT"
>
  <label
    className="Label-vlikyy-0 bZPWgf"
    htmlFor="swaps"
  >
    Swap against (seperate by comma)
  </label>
  <div
    className="SwapTags__StyledTagInput-sc-15dr4e3-3 dFkJJT"
  >
    <div
      className="SwapTags__StyledTags-sc-15dr4e3-1 jDvkBJ"
    >
      <span
        className="Tags__SwapTag-uz6t9v-1 heiBUy"
      >
        test
         
        <b
          className="SwapTags__StyledDelete-sc-15dr4e3-2 daIZly"
          onClick={[Function]}
        >
          x
        </b>
      </span>
    </div>
    <input
      className="SwapTags__StyledTagInputField-sc-15dr4e3-4 hcXjpB"
      id="swaps"
      name="swaps"
      onInput={[Function]}
      placeholder="type new tag here..."
    />
  </div>
</section>
`;
